File: basic.java
import java.util.*;
import java.util.stream.*;

public class ProofreadingTool {

    // 1. isPalindrome
    public static boolean isPalindrome(String word) {
        if (word == null) return false;
        String cleaned = word.toLowerCase();
        return cleaned.equals(new StringBuilder(cleaned).reverse().toString());
    }

    // 2. findPalindromeBreaks
    public static List<Integer> findPalindromeBreaks(List<String> words) {
        if (words == null || words.isEmpty()) return new ArrayList<>();
        List<Integer> breaks = new ArrayList<>();
        for (int i = 0; i < words.size(); i++) {
            if (!isPalindrome(words.get(i))) {
                breaks.add(i);
            }
        }
        return breaks;
    }

    // 3. findRepeatedPhrases
    public static List<Integer> findRepeatedPhrases(List<String> words, int phraseLength) {
        if (words == null || phraseLength >= words.size() || phraseLength <= 0) {
            return new ArrayList<>();
        }
        
        Map<String, List<Integer>> phraseMap = new HashMap<>();
        
        for (int i = 0; i <= words.size() - phraseLength; i++) {
            List<String> sub = words.subList(i, i + phraseLength);
            String phrase = String.join(" ", sub).toLowerCase();
            
            phraseMap.putIfAbsent(phrase, new ArrayList<>());
            phraseMap.get(phrase).add(i);
        }
        
        List<Integer> result = new ArrayList<>();
        for (List<Integer> indices : phraseMap.values()) {
            if (indices.size() > 1) {
                result.addAll(indices);
            }
        }
        
        Collections.sort(result);
        return result;
    }

    // 4. analyzeTexts
    public static List<Map<String, Object>> analyzeTexts(List<List<String>> texts, int phraseLength) {
        if (texts == null || texts.isEmpty()) return new ArrayList<>();
        
        List<Map<String, Object>> result = new ArrayList<>();
        
        for (List<String> text : texts) {
            Map<String, Object> analysis = new HashMap<>();
            analysis.put("repeatedPhrases", findRepeatedPhrases(text, phraseLength));
            analysis.put("palindromeBreaks", findPalindromeBreaks(text));
            result.add(analysis);
        }
        
        return result;
    }
}

File: basic.js
// 1. isPalindrome
function isPalindrome(word) {
  if (typeof word !== 'string') return false;
  const cleaned = word.toLowerCase();
  return cleaned === cleaned.split('').reverse().join('');
}

// 2. findPalindromeBreaks
function findPalindromeBreaks(words) {
  if (!Array.isArray(words) || words.length === 0) return [];
  const breaks = [];
  for (let i = 0; i < words.length; i++) {
    if (!isPalindrome(words[i])) {
      breaks.push(i);
    }
  }
  return breaks;
}

// 3. findRepeatedPhrases
function findRepeatedPhrases(words, phraseLength) {
  if (!Array.isArray(words) || phraseLength >= words.length || phraseLength <= 0) {
    return [];
  }
  
  const phraseMap = new Map();
  const result = [];
  
  for (let i = 0; i <= words.length - phraseLength; i++) {
    const phrase = words.slice(i, i + phraseLength).join(' ').toLowerCase();
    
    if (!phraseMap.has(phrase)) {
      phraseMap.set(phrase, []);
    }
    phraseMap.get(phrase).push(i);
  }
  
  for (const indices of phraseMap.values()) {
    if (indices.length > 1) {
      result.push(...indices);
    }
  }
  
  return result.sort((a, b) => a - b);
}

// 4. analyzeTexts
function analyzeTexts(texts, phraseLength) {
  if (!Array.isArray(texts) || texts.length === 0) return [];
  
  return texts.map(text => ({
    repeatedPhrases: findRepeatedPhrases(text, phraseLength),
    palindromeBreaks: findPalindromeBreaks(text)
  }));
}
